homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,447 @@
+"""Support for Nexia / Trane XL thermostats."""
+import logging
+
+from nexia.const import (
+    FAN_MODES,
+    OPERATION_MODE_AUTO,
+    OPERATION_MODE_COOL,
+    OPERATION_MODE_HEAT,
+    OPERATION_MODE_OFF,
+    OPERATION_MODES,
+    SYSTEM_STATUS_COOL,
+    SYSTEM_STATUS_HEAT,
+    SYSTEM_STATUS_IDLE,
+    UNIT_FAHRENHEIT,
+)
+import voluptuous as vol
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_AUX_HEAT,
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HUMIDITY,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    ATTR_TARGET_TEMP_STEP,
+    CURRENT_HVAC_COOL,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_AUX_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    STATE_OFF,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AIRCLEANER_MODE,
+    ATTR_DEHUMIDIFY_SETPOINT,
+    ATTR_DEHUMIDIFY_SUPPORTED,
+    ATTR_HOLD_MODES,
+    ATTR_HUMIDIFY_SETPOINT,
+    ATTR_HUMIDIFY_SUPPORTED,
+    ATTR_SETPOINT_STATUS,
+    ATTR_ZONE_STATUS,
+    ATTRIBUTION,
+    DATA_NEXIA,
+    DOMAIN,
+    MANUFACTURER,
+    NEXIA_DEVICE,
+    UPDATE_COORDINATOR,
+)
+from .entity import NexiaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_SET_AIRCLEANER_MODE = "set_aircleaner_mode"
```

Comment:
Where are these services registered?

Reply:
I removed them from this PR to cut down the size and plan on adding them back later. I missed some cruft.  removing now